cs/goteam/README.md

```diff
@@ -0,0 +1,280 @@
+
+
+## Jmenné konvence - Balíčky
+
+Název balíčku by měl mít tyto vlastnosti:
+
+- je krátký
+
+	Raději "transport" než "transportmechanisms" (transportní mechanismus)
+
+- jasný
+
+    Jméno objasňující funkci balíčku: "bytes"
+    Jméno popisující implementaci externí závislosti: "postgres"
+
+Balíčky by měly mít jednoznačnou funkcionalitu. Vyhněte se *všeřešícím* balíčkům:
+
+	util (nástroje)
+	helpers (pomocné funkce)
+	atd.
+
+Často je to pouze znamením, že vám někde chybí interface.
+
+    util.ConvertOtherToThing() (změň tohle na věc) by měl být refaktorován na interface pro *Věc*
+
+*všeřešící* balíčky budou pravděpodobně prvním místem, kde začnete narážet na problémy při testování či kruhových závislostech.
+
+## Jmenné konvence - Proměnné
+
+Mezi běžné konvence pro pojmenování proměnných patří:
+
+- používejte camelCase ne snake_case
+- pro indexy používejte jednopísmenné proměnné
+	- for i:=0; i < 10; i++ {}
+- používejte krátké ale popisné názvy pro pojmenování všeho ostatního
+	- var count int (var pocet int)
+	- var cust Customer (var zak Zakaznik)
+
+Používáním zbytečně krátkých jmen pro proměnné nic nezískáte. K pojmenování můžete použít jednoduchý princip:
+
+    Čím dále od deklarace používáte proměnnou tím delší (popisnější) by její název měl být.
+
+Zde pár příkladů pojmenování proměnné na základě kontextu:
+
+.link https://github.com/golang/go/blob/master/src/fmt/format.go#L12
+.link https://github.com/golang/go/blob/master/src/fmt/format.go#L326
+.link https://github.com/golang/go/blob/master/src/flag/flag.go#L293
+
+- pro pojmenování pole prvků použijte opakující se písmeno
+	- var tt []*Thing (var vv []*Věc)
+
+- uvnitř cyklu používejte jednopísmenné proměnné
+	- for i, t := range tt {}
+
+Tyto konvence se běžně používají i v samotném zdrojovém kódu Go.
+
+## Jmenné konvence - funkce a metody
+
+Funkcním v balíčku nepřidávejte do názvu jméno balíčku:
+
+	SPRÁVNĚ:  log.Info()
+	ŠPATNĚ:   log.LogInfo()
+
+Samotný název balíčku definuje jeho účel, není třeba jej opakovat.
+
+Go kód neobsahuje settery ani gettery.
+
+	SPRÁVNĚ:  custSvc.Customer()
+	ŠPATNĚ:   custSvc.GetCustomer()
+
+## Jmenné konvence - Interfaces (rozhraní)
+
+Pokud váš interface obsahuje pouze jednu funkci, připojte za její jméno "-er":
+
+	type Stringer interface{ //Textovač
+		String() string //Text
+	}
+
+Pokud má váš interface více funkcí, pojmenujte ho tak aby popisoval svou funkcionalitu:
+
+	type CustomerStorage interface { //zákaznické úložiště
+		Customer(id int) (*Customer, error) //zákazník
+		Save(c *Customer)  error //uložit
+		Delete(id int) error //smazat
+	}
+
+Puritáni by mohli namítnout, že všechny interface by měli mít koncovku `-er`. Dle mého názoru je popisnost a čitelnost přednější.
+
+## Jmenné konvence - Zdrojový kód
+
+Uvnitř balíčku oddělujte kód do logických celků.
+
+Pokud balíček pracuje s více objekty, ponechte logiku pro každý objekt ve vlastním souboru:
+
+	src/goteam/cviceni/sklad/postgres
```

Comment:
please, do not translate the paths, they are really paths for code available for download. same goes for filenames